analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.cs

```diff
@@ -290,6 +293,54 @@ public void ToString_WithAll()
 ");
         }
 
+        [TestMethod]
+        public void ResetFieldConstraints_NotPreservesField() =>
+            ResetFieldConstraintTests(ObjectConstraint.Null, false);
+
+        [TestMethod]
+        public void ResetFieldConstraints_PreservesField() =>
+            ResetFieldConstraintTests(LockConstraint.Held, true);
+
+        [TestMethod]
+        public void ResetFieldConstraints_ResetFieldsAsDefined()
+        {
+            var instanceField = CreateFieldSymbol("object field;");
+            var staticField = CreateFieldSymbol("static object field;");
+
+            var preserveAll = LockConstraint.Held;
+            var preserveNone = ObjectConstraint.Null;
+
+            var sut = ProgramState.Empty;
+            sut = sut.SetSymbolValue(instanceField, new SymbolicValue().WithConstraint(preserveAll).WithConstraint(preserveNone))
+                     .SetSymbolValue(staticField, new SymbolicValue().WithConstraint(preserveAll).WithConstraint(preserveNone));
+            sut = sut.ResetFieldConstraints();
+            var instanceFieldSymbolValue = sut[instanceField];
+            var staticFieldSymbolValue = sut[staticField];
+            instanceFieldSymbolValue.HasConstraint(preserveAll).Should().BeTrue(because: "preserveAll should be preserved for instance fields");
```

Comment:
4x here
```suggestion
            instanceFieldSymbolValue.HasConstraint(preserveAll).Should().BeTrue("preserveAll should be preserved for instance fields");
```

Reply:
Actually the assertion line itself is pretty clear to me. I'd remove the `because` explanations.